src/pages/auth.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { useLocation, Redirect } from "@docusaurus/router";
+import Layout from "@theme/Layout";
+import { API_URL } from "../constants";
+
+const Auth = () => {
+  const location = useLocation();
+  const params = new URLSearchParams(location.search);
+  const [count, setCount] = React.useState(3);
+  const [loading, setLoading] = React.useState(true);
+
+  const stateString = params.get("state");
+
+  const redirect = (url) => <Redirect to={url} />;
```

Comment:
No, you are not missing anything. I think I did it that way because it was going to be called in few places. But extracting to a separate component makes more sense